cli/tests/config.go

```diff
@@ -0,0 +1,125 @@
+// Package tests provides helper utilities for CLI E2E/integration tests.
+// This file implements a builder that produces a CLI config.json file at a
+// caller-specified location (or a temporary file when no location is given).
+// The shape of the JSON mirrors the fields consumed by cli/internal/config.
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+// ConfigBuilder helps construct a Rudder CLI configuration file programmatically
+// for test cases.
+//
+// Usage example:
+//
+//	cfg := tests.NewConfigBuilder().
+//	        WithAccessToken("tok_123").
+//	        WithTelemetryEnabled("write-key", "https://telemetry.example.com").
+//	        Build()
+//
+// The Build method returns the path to the generated config file, a cleanup
+// function, and an error if something went wrong.
+//
+// All methods mutate the builder in-place and return the same instance,
+// allowing chaining.
+// NOTE: Only the fields relevant to current tests are implemented. Extend as
+// needed to cover additional configuration keys.
+type ConfigBuilder struct {
```

Comment:
can we create ConfigBuilder as a composite type that consits of the actual Config struct and the filename
```suggestion
type ConfigBuilder struct {
    Config
    filename string
}
````

Reply:
We could add builder functions like `WithConfig` as and when required by the tests

Reply:
I would actually try to keep things simple here, only include config properties that we actually want to include during e2e tests. Otherwise, the ConfigBuilder will start to replicate work that would feel more natural to the config package itself, i.e creating a config file (or its contents) from Config data.

I would also like to mention here again that I would prefer the access token (and maybe api url as well) out of the config file if we're already expecting those to be passed through environmental variables to the test suite. We don't need to test that config is parsed correctly by the CLI, that can be done with unit tests.